src/main/java/com/example/demo/contrtoller/BookController.java

```diff
@@ -48,4 +50,12 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestD
     public void softDeleteBook(@PathVariable Long id) {
         bookService.softDeleteBookById(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> search(@RequestParam(required = false) String titlePart,
+                                @RequestParam(required = false) String authorPart) {
+        BookSearchParametersDto searchParameters = new BookSearchParametersDto(
+                titlePart, authorPart);
```

Comment:
```suggestion
    public List<BookDto> search(BookSearchParametersDto searchParameters) 
```
you can pass searchParameters directly